src/com/twu/biblioteca/MainMenu.java

```diff
@@ -1,40 +1,62 @@
 package com.twu.biblioteca;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Scanner;
 
 public class MainMenu {
 
-    public void welcome() {
-        System.out.println("Welcome to Biblioteca. Your one-stop-shop for great book titles in Bangalore!");
+    private final String welcome = "Welcome to Biblioteca. Your one-stop-shop for great book titles in Bangalore!";
+    private final String invalidMessage = "Please select a valid option!";
+    private final String goodBye = "Goodbye!";
+
+   ArrayList<String> books = new ArrayList<String>(Arrays.asList(
```

Comment:
I'm curious to know what caused this refactoring... was it looking ahead at the 1.7 story?

Reply:
A little bit, as I had to learn how to do it and it would be useful for future stories. The main reason was your comment about maintaining the messages in the future.